src/utils/delegation.ts

```diff
@@ -0,0 +1,113 @@
+import { subgraphRequest } from '../utils';
+import delegationSubgraphs from '../delegationSubgraphs.json';
+
+export const SNAPSHOT_SUBGRAPH_URL = delegationSubgraphs;
+const PAGE_SIZE = 1000;
+
+type Delegation = {
+  delegator: string;
+  delegate: string;
+  space: string;
+  timestamp: number;
+};
+
+export default async function getDelegatesBySpace(
+  network: string,
+  space: string,
+  snapshot: string | number = 'latest',
+  options: any = {}
+) {
+  const subgraphUrl = options.subgraphUrl || SNAPSHOT_SUBGRAPH_URL[network];
+  if (!subgraphUrl) {
+    return Promise.reject(
+      `Delegation subgraph not available for network ${network}`
+    );
+  }
+
+  let result: Delegation[] = [];
+  const spaceIn = ['', space];
+  if (space.includes('.eth')) spaceIn.push(space.replace('.eth', ''));
+
+  while (true) {
+    const newResults = await fetchData({
+      url: subgraphUrl,
+      spaces: spaceIn,
+      pivot: result[result.length - 1]?.timestamp || 0,
+      snapshot
+    });
+
+    result = mergeWithoutDuplicates(result, newResults);
+
+    if (
+      newResults.length === PAGE_SIZE &&
+      newResults[0].timestamp === newResults[newResults.length - 1].timestamp
+    ) {
+      throw new Error('Unable to paginate delegation');
+    }
+
+    if (newResults.length < PAGE_SIZE) break;
+  }
+
+  return result;
+}
+
+export function mergeWithoutDuplicates(a: Delegation[], b: Delegation[]) {
```

Comment:
`a` and `b` are too confusing, maybe we can use something like `existing` and `new` 😄

Reply:
There's no concept of new or ancient in this case, it's returning same thing no matter the order